### ISSUE TITLE
It shouldn't normalise the dim at every search

### ISSUE THREAD

**Issue by tttp** [dc-js/dc.js#1464]
Hi,
I've switched to use the code put into the core, yeah!

However (and it might completely be an issue in my original code), it normalises each item for everysearch, but it should have been done in the dimension already, so it's slowing down things for no reason and can be simplified to d.indexOf(query) !== -1

```
function (query) {
    query = _normalize(query);
    return function (d) {
        return _normalize(d).indexOf(query) !== -1;
    };
};
```

also, I've added a debounce so it doesn't filter/redraw after each char when you type fast, is this on purpose that it was removed? I think it's working well and no downside, and could be added in the core.

I got a dev dc working again, you can assign me this issue if you want me to fix it ;)

**Comment by gordonwoodhull** [comment]
Huh, that's weird. @kum-deepak, is it possible that you based your PR off an earlier version of `filter-input.js` or missed some later commits in the squash? A few functions got removed in f20c7b0410c8a9; it doesn't look like an ordinary refactor.

**Comment by kum-deepak** [comment]
Let me check with my notes 😄

**Comment by tttp** [comment]
Hi @kum-deepak,

Yes, indeed, you search on a normalized data. this is what building the dimension is made for ;)
and agree, it has to be made explicit: you need to run the normalisation. May be instead of toLowerCase, the example should use the normalize() function?

But I strongly think that it makes sense to normalize once, when you build the dimension. As you mention, I've been working on datasets that are more complex than "simply" lowercase, eg german names where you you can use ß or ss, or joerg and jörg are seen at the same. That make sense to do only once.

I think that a flag isn't needed and we simply need to make it explicit in the doc, but willing to be convinced of the opposite ;)

I completely see your point about debounce (actually, throttle as I implemented it I think)... but this is an internal design: it needs to throttle, it would be way better to use a common implementation, but in the meanwhile, it makes sense to keep it only in this graph, and switch to the generic implementation when done, isn't it?

As in, from a user point of view, it works the same, it's an implementation detail. Without the throttle, I'd be force to code it outside of the widget, and then I'd be fighting against dc once it's implemented in the core

But if you think this use case is what is needed to get a generic implementation and we can get it soon, happy to +1 on waiting and adding the generic version on this graph ;) It depends on what is short term for in your mind ;)

X+

**Comment by kum-deepak** [comment]
Thanks @tttp!

Looks like we are converging, it would be great if you could raise 2 PRs (if possible), one covering the normalization, and other for the throttle. This will ease the work in future when dc moves to standard implementation of throttle/debounce.

Also would be nice to have an example using German data and associated normalization (can be in the same example).
